roles/puppet_repositories/vars/main.yml

```
@@ -0,0 +1 @@
+puppet_repositories_supported_versions: [5, 6]
```

```suggestion
puppet_repositories_supported_versions:
  - 5
  - 6
```

I prefer long lists, but that's not what my brain jumped at me.

You use integers here, but nothing forbids the user to pass a string as the desired version, and then the `difference` above will not do what we want:

```console
% ansible -m debug -a 'msg={{ [1,2] | difference(["2"]) }}' localhost
localhost | SUCCESS => {
    "msg": [
        1,
        2
    ]
}
```

So I guess we need to hard-convert things to strings at some point. But IMHO this can go as an improvement in a later PR.